x/pairing/keeper/msg_server_relay_payment.go

```diff
@@ -271,6 +294,9 @@ func (k msgServer) RelayPayment(goCtx context.Context, msg *types.MsgRelayPaymen
 		rejected_relays_num--
 	}
 
+	// save all payment objects
+	paymentHandler.Flush(ctx)
```

Comment:
wouldn't it be better if we put the flush after the `rejected_relays_num != 0` check? In other words, we don't wanna update the KVStore for a failed relay TX